src/material.h

```diff
@@ -29,6 +30,9 @@ class material_type
         int _fire_resist;
         int _chip_resist;       // Resistance to physical damage of the item itself
         int _density;   // relative to "powder", which is 1
+        bool _edible;
+
+        mutable std::map<vitamin_id, double> _vitamins;
```

Comment:
> Got any source on that?

`... specifies that the member does not affect the externally visible state of the class (as often used for mutexes, memo caches, lazy evaluation, and access instrumentation). mutable members of const classes are modifiable. ` [source](http://en.cppreference.com/w/cpp/language/cv)

> It looks really ugly to allow modification of the map just to use the non-const operator on it, instead of just using a const-correct way of accessing the value.

It could be changed to the `find` then `iter != end` idiom?